### PR TITLE
Update docs with fresher bazel example

### LINKED ISSUE #3410: Update docs with fresher bazel support.
Hi there!
Right now it is quite a lot of work to get PyO3 working with Bazel, but it is possible. It took me many hours of work to get this setup, and most existing advice online is very out of date, or I couldn't get it to work.

There is an closed issue which links to a repo with bazel rules for PyO3:  https://github.com/PyO3/pyo3/issues/481

However, these rules are pretty outdated and do not work with the current versions of rules_rust.
I would suggest adding a link to my repo where I have got a working example of using PyO3 0.19.2 with rules_rust and python 3.9.

If I find the time, I may try and consolidate this to form a new `rule_pyo3` repo.

[commit with working pyo3 and bazel](https://github.com/OliverFM/pytorch_with_gazelle/commit/b65c35c7bfc57107b61591a6c8222edf778ffbd7).

I would be very happy to make a PR with these changes to the docs.

### PR DESCRIPTION
**Pull request by OliverFM**
Adds docs to fix #3410

### CODE REVIEW

**Comment by OliverFM** [comment]
Any suggestions on getting the CI to pass? I can't make sense of the error, or see why my changes would've triggered it.

**Comment by davidhewitt** [comment]
Think a recent upstream change has broken us, I hope to try to debug later

File: newsfragments/3411.added.md
Comment (davidhewitt) [inline newsfragments/3411.added.md:1]: No need to have docs updates in the CHANGELOG, it'll be live as soon as we merge. Can remove this.

File: guide/src/building_and_distribution.md [hunk 1]
```diff
@@ -88,7 +88,12 @@ You can then open a Python shell in the output directory and you'll be able to r
 
 If you're packaging your library for redistribution, you should indicated the Python interpreter your library is compiled for by including the [platform tag](#platform-tags) in its name. This prevents incompatible interpreters from trying to import your library. If you're compiling for PyPy you *must* include the platform tag, or PyPy will ignore the module.
 
-See, as an example, Bazel rules to build PyO3 on Linux at https://github.com/TheButlah/rules_pyo3.
+#### Bazel builds
+
+To use PyO3 with bazel one needs to manually PyO3, PyO3-ffi and PyO3-macros. In particular, one needs to make sure that it is compiled with the right python flags for the version you intend to use.
```
Comment (davidhewitt) [inline guide/src/building_and_distribution.md:93]: I think a word is missing here?

```suggestion
To use PyO3 with bazel one needs to manually configure PyO3, PyO3-ffi and PyO3-macros. In particular, one needs to make sure that it is compiled with the right python flags for the version you intend to use.
```

**Review by davidhewitt — COMMENTED** [review]
Thanks, looks great with just a couple small adjustments.

If you rebase CI should be happy now.

File: newsfragments/3411.added.md
Comment (OliverFM) [reply newsfragments/3411.added.md:1]: Removed!

**Review by davidhewitt — APPROVED** [review]
Thanks, I'll squash and merge later (or if you squash first, I'll click merge). 👍

**Pull request merged**